src/fsharp/FSharp.Core/array.fs

```diff
@@ -957,9 +957,13 @@ namespace Microsoft.FSharp.Collections
             accv
 
         [<CompiledName("Average")>]
-        let inline average      (array:_[]) = 
+        let inline average      (array:'T[]) = 
             checkNonNull "array" array
-            Seq.average array
+            if array.Length = 0 then invalidArg "array" LanguagePrimitives.ErrorStrings.InputArrayEmptyString
```

Comment:
Let me know how to show this exactly. I have read the contributor guidance and I do know that I have not provided the numbers. But I think that the the current implementation was just an omission or something given the `Sum` implementation is already there. Have spent almost 30 min starring at code why FsCheck thinks `average` is not conclusive in the 1 of 3 tests. Will spend more tomorrow to set up a proper testing environment on my remote PC.